wire-ios/Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Advanced.swift

```diff
@@ -43,11 +43,19 @@ extension SettingsCellDescriptorFactory {
     }
 
     // MARK: - Sections
-    private var troubleshootingSection: SettingsSectionDescriptor {
+    private func troubleshootingSection(userSession: UserSession) -> SettingsSectionDescriptor {
         let submitDebugButton = SettingsExternalScreenCellDescriptor(
             title: SelfSettingsAdvancedLocale.Troubleshooting.SubmitDebug.title,
             presentationAction: { () -> (UIViewController?) in
-                return SettingsTechnicalReportViewController()
+                let router = SettingsDebugReportRouter()
+                let viewModel = SettingsDebugReportViewModel(
+                    router: router,
+                    shareFile: userSession.shareFileUseCase,
```

Comment:
Actually we have discussed recently that we don't want to provide use cases through the user session, as it leads to a uncontrollable amount of dependencies.

You can create the use case right here in place with the dependencies from user session, like:
`ShareFileUseCase(contextProvider: userSession)` or `ShareFileUseCase(contextProvider: userSession.contextProvider)` (not sure what is correct right now).

Reply:
done https://github.com/wireapp/wire-ios/pull/1658/commits/8c85d9ddbe3d4f96ae06fb91a9db48865e963819